### PR TITLE
Feat/2280 remove features native and web in favor of target arch

### LINKED ISSUE #2280: Remove features `native` and `web` in favor of `target_arch`

### PR DESCRIPTION
**Pull request by FirelightFlagboy**
close #2280

### CODE REVIEW

File: oxidation/libparsec_python/src/lib.rs [hunk 1]
```diff
@@ -1,5 +1,7 @@
 // Parsec Cloud (https://parsec.cloud) Copyright (c) BSLv1.1 (eventually AGPLv3) 2016-2021 Scille SAS
```
Comment (touilleMan) [inline oxidation/libparsec_python/src/lib.rs:2]: libparsec_python will never be compiled for wasm32 so no need for this

File: oxidation/libparsec_python/Cargo.toml [hunk 2]
```diff
@@ -23,7 +23,7 @@ uuid = { version = "1.0.0-alpha.1", features = ["serde", "v4", "fast-rng"] }
 chrono = {version = "0.4.19", features = ["serde"]}
 fancy-regex = "0.8.0"
 
-[dependencies.pyo3]
+[target.'cfg(not(target_arch = "wasm32"))'.dependencies.pyo3]
```
Comment (touilleMan) [inline oxidation/libparsec_python/Cargo.toml:26]: libparsec_python will never be compiled for wasm32 so no need for this

**Review by touilleMan — APPROVED** [review]

File: oxidation/libparsec_python/src/lib.rs (hunk 1 above)
Comment (FirelightFlagboy) [reply oxidation/libparsec_python/src/lib.rs:2]: I've added a comment

File: oxidation/libparsec_python/Cargo.toml (hunk 2 above)
Comment (FirelightFlagboy) [reply oxidation/libparsec_python/Cargo.toml:28]: I've added a comment

**Review by TimeEngineer — APPROVED** [review]

**Pull request merged**